modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -1297,4 +1316,165 @@ public static boolean shouldSwitchToRequestsExecutor(ThreadOperation... required
             return true;
         }
     }
+
+    /**
+     * Performs thread dump and prints all available info to the given log with WARN logging level.
+     *
+     * @param log Logger.
+     */
+    public static void dumpThreads(IgniteLogger log) {
```

Comment:
Do we want to have this method in the overly generic `IgniteUtils`? How about creating something like `ThreadUtils` for it? In AI2 `IgniteUtils` is so huge that it makes IDEs work really slow, navigation is pretty difficult. I think we should modularize this.

Reply:
I thought exactly the same but did not do it... Will create `ThreadUtils`. Thank you for the suggestion/support!